### DISCUSSION TITLE
Error when compiling

### DISCUSSION THREAD

**Discussion by tudorels** [Arnold-n/P1P2MQTT#58] [Q&A] [answered]
Hello, I try to compile the code and I get error:
/P1P2Serial/examples/P1P2-bridge-esp8266/P1P2-bridge-esp8266.ino: In function 'void process_for_mqtt_json(byte*, int)':
P1P2-bridge-esp8266:1465:17: error: 'bytes2keyvalue' was not declared in this scope
 1465 |     if (n == 3) bytes2keyvalue(rb[0], rb[2], EMPTY_PAYLOAD, rb + 3, mqtt_key, mqtt_value);
      |                 ^~~~~~~~~~~~~~
P1P2-bridge-esp8266:1469:23: error: 'bytes2keyvalue' was not declared in this scope
 1469 |         int kvrbyte = bytes2keyvalue(rb[0], rb[2], i - 3, rb + 3, mqtt_key, mqtt_value);
      |                       ^~~~~~~~~~~~~~
P1P2-bridge-esp8266:1491:40: error: 'bits2keyvalue' was not declared in this scope
 1491 |             int kvr = (kvrbyte == 8) ? bits2keyvalue(rb[0], rb[2], i - 3, rb + 3, mqtt_key, mqtt_value, j) : kvrbyte;
      |                                        ^~~~~~~~~~~~~
exit status 1
'bytes2keyvalue' was not declared in this scope

I'm doing something wrong? What can I do to compile correctly? Thank you!

**Comment by Arnold-n** [comment, marked as answer]
Please uncomment one of lines 55 or 56 in P1P2_Config.h to match your Daikin system:
```
//#define E_SERIES
//#define F_SERIES
```

**Comment by tudorels** [comment]
It works! Thank you!

**Comment by Arnold-n** [comment]
Great, I will add an #error warning in the next version that is more informative than  the compiler error you received earlier.
